System.NullReferenceException from AL task in Microsoft.Build.Tasks.Core

<!-- This is a template that helps us provide quicker feedback. Please use any relevant sections and delete anything you don't need. -->

### Issue Description
We're hitting this build error from some projects in VS 2022 **Int Preview** (both inside VS and from command line)
    MSB4018: The "AL" task failed unexpectedly with System.NullReferenceException

The same projects build well in VS 2022 **Current** 
### Steps to Reproduce
You can repro it by building [Platform.retail.sln ](https://dev.azure.com/devdiv/devdiv/_git/VS?path=/src/Platform/Platform.retail.sln)

or build this project e.g. [Microsoft.VisualStudio.Utilities.csproj](https://dev.azure.com/devdiv/devdiv/_git/VS?path=/src/Platform/Utilities/Impl/Microsoft.VisualStudio.Utilities.csproj)

### Analysis
It looks like some regression on MSBuild, since the same projects build well in VS 2022 **Current** (the shipped version).

Below is from the binlog
C:\Program Files\Microsoft Visual Studio\2022\Preview\MSBuild\Current\Bin\amd64\Microsoft.Common.CurrentVersion.targets(3870,5): error MSB4018: The "AL" task failed unexpectedly.
System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.Build.Tasks.AL.GenerateFullPathToTool()
   at Microsoft.Build.Utilities.ToolTask.ComputePathToTool()
   at Microsoft.Build.Utilities.ToolTask.Execute()
   at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()
   at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext() [D:\DD\VS-MSB\src\Platform\Utilities\Impl\Microsoft.VisualStudio.Utilities.csproj]

### Attach a binlog
Here's the [msbuild.binlog](https://microsoft-my.sharepoint.com/:u:/p/hlong/EVIk3mZnzppHtSFiAD9Mm0wBg9jzha3GqEJ0mZttFL85GA?e=aGNnXx)

--- a/src/Tasks/Al.cs
+++ b/src/Tasks/Al.cs
@@ -306,8 +306,9 @@ namespace Microsoft.Build.Tasks
             if (String.IsNullOrEmpty(pathToTool) || !FileSystems.Default.FileExists(pathToTool))
             {
                 // The bitness of al.exe should match the platform being built
-                string archToLookFor =  Platform.Equals("x86", StringComparison.OrdinalIgnoreCase) ? Platform :
-                                        Platform.Equals("x64", StringComparison.OrdinalIgnoreCase) ? ProcessorArchitecture.AMD64 : // x64 maps to AMD64 in GeneratePathToTool
+                // Yoda condition prevents null reference exception if Platform is null.
+                string archToLookFor =  "x86".Equals(Platform, StringComparison.OrdinalIgnoreCase) ? Platform :
+                                        "x64".Equals(Platform, StringComparison.OrdinalIgnoreCase) ? ProcessorArchitecture.AMD64 : // x64 maps to AMD64 in GeneratePathToTool
                                         ProcessorArchitecture.CurrentProcessArchitecture;
 
                 pathToTool = SdkToolsPathUtility.GeneratePathToTool(f => SdkToolsPathUtility.FileInfoExists(f), archToLookFor, SdkToolsPath, ToolExe, Log, true);

